common/db.go

```diff
@@ -38,14 +44,34 @@ func InitDB(dbPath string) (*DB, error) {
 	d.dbh = db
 	schemaBuf := new(bytes.Buffer)
 	tmpl := template.Must(template.New("schema").Parse(schema))
-	tmpl.Execute(schemaBuf, nil)
+	tmpl.Execute(schemaBuf, LivepeerDBVersion)
 	_, err = db.Exec(schemaBuf.String())
 	if err != nil {
 		glog.Error("Error initializing schema ", err)
 		d.Close()
 		return nil, err
 	}
 
+	// Check for correct DB version and upgrade if needed
+	var dbVersion int
+	row := db.QueryRow("SELECT value FROM kv WHERE key = 'dbVersion'")
+	err = row.Scan(&dbVersion)
+	if err != nil {
+		glog.Error("Unable to fetch DB version ", err)
+		d.Close()
+		return nil, err
+	}
+	if dbVersion > LivepeerDBVersion {
+		glog.Error("Database too new")
+		d.Close()
+		return nil, ErrDBTooNew
+	} else if dbVersion < LivepeerDBVersion {
+		// Upgrade stepwise up to the correct version using the migration
+		// procedure for each version
+	} else if dbVersion == LivepeerDBVersion {
+		// all good; nothing to do
```

Comment:
Updated this comment; hopefully it's clearer that this no-op is expected @yondonfu